### DISCUSSION TITLE
Can we create username and password login and saml sso on the same client ?

### DISCUSSION THREAD

**Discussion by dprasanthv** [keycloak/keycloak#26000] [Q&A] [answered]
I am trying to setup a saml idp along with username and password form for my website. Users can login using both username & password or saml. I configured saml idp with First Broker login, it ends up as "Your login attempt timed out.  Login will start from the beginning.". When saml is configured as Required, all the logins end up redirecting to saml login without an option for username and password form in the **browser** flow. Any help appreciated

**Comment by darius-m** [comment]
How will users access the site? Usually having both username + password and SAML on a site usually means that you expect to have some accounts that are local or don't rely on SSO (in the database, or maybe LDAP / AD), while others (or most) accounts should log in through SAML (via Keycloak). Do you have a different scenario?

**Reply by dprasanthv** [reply]
Users belonged to same entity are grouped under an organization (external). There will be many organizations, which will have their own sso/saml, we are implementing an spi to extract the domain of the email address provided and redirect the user to their respective SSO URL. When any user of that specific organization tries to login via saml/sso, they will be redirected to their idp and remaining users who are not part of any organization login via their username and password.

**Reply by dprasanthv** [reply, marked as answer]
Used provider login for multiple samls along with username and password did the magic. Thanks for looking
